### PR TITLE
Dart + docker samples

### PR DESCRIPTION
**Pull request by kevmoo**
One basic shelf sample
One more complex sample demonstrating Google APIs from Cloud Run

Both include Dockerfiles

### CODE REVIEW

File: server/google_apis/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+A Dart HTTP services meant to be deployed to to
```
Comment (mit-mit) [inline server/google_apis/README.md:1]: s/HTTP services/HTTP service/

File: server/google_apis/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Use Google's official Dart image.
```
Comment (mit-mit) [inline server/google_apis/Dockerfile:1]: Isn't this identical to the `dart-runtime` image?

File: server/google_apis/mono_pkg.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+dart:
+  - dev
+  - 2.12.0
+
+stages:
+  - analyze:
+    - dartanalyzer: --fatal-infos .
```
Comment (mit-mit) [inline server/google_apis/mono_pkg.yaml:7]: Doesn't mono_repo support generating these with the `dart` commands?

File: server/google_apis/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+name: cloud_run_google_apis
+publish_to: none # let's not accidentally publish this to pub.dev
```
Comment (mit-mit) [inline server/google_apis/pubspec.yaml:2]: I think I'd s/let's not accidentally publish this to pub.dev/not intended for publication to pub.dev/

**Review by domesticmouse — APPROVED** [review]
Next time, can we separate the generated guff from the actual new sample code? I'm really unsure what is template and what is real.

File: server/google_apis/Dockerfile (hunk 2 above)
Comment (kevmoo) [reply server/google_apis/Dockerfile:1]: No! See https://github.com/dart-lang/dart_docker/issues/69

It really matters when doing the default deploy w/ cloud run!

File: server/google_apis/mono_pkg.yaml (hunk 3 above)
Comment (kevmoo) [reply server/google_apis/mono_pkg.yaml:7]: It's complicated.

This syntax is specific to `mono_repo`, which started out only supporting travis ... which was based on the old way of doing things.

No high on my list to fix.

File: server/google_apis/README.md (hunk 1 above)
Comment (kevmoo) [reply server/google_apis/README.md:1]: ack

File: server/google_apis/pubspec.yaml (hunk 4 above)
Comment (kevmoo) [reply server/google_apis/pubspec.yaml:2]: ack

**Comment by kevmoo** [comment]
> Next time, can we separate the generated guff from the actual new sample code? I'm really unsure what is template and what is real.

What do you mean? The generated bits are just the github actions stuff – which goes along w/ this PR. So it's tested.

Anything else @domesticmouse ?

**Pull request merged**